test/task.h

```diff
@@ -193,10 +209,10 @@ typedef enum {
 #define ASSERT_UINT64_NE(a, b) ASSERT_INT_BASE(a, !=, b, uint64_t, PRIu64)
 
 #define ASSERT_STR_EQ(a, b) \
-  ASSERT_BASE(strcmp(a, b) == 0, a, ==, b, s)
+  ASSERT_BASE(strcmp(a, b) == 0, a, ==, b, char*, s)
```

Comment:
Just pass `"s"` (and `"p"`) to `ASSERT_BASE? No need for a separate ASSERT_INT_BASE macro.

Reply:
Yeah, much nicer now. Thanks!